mmv1/products/compute/InstanceGroupMembership.yaml

```diff
@@ -0,0 +1,106 @@
+
+  **NOTE** This resource has been added to avoid a situation, where after
+  Instance is recreated, it's removed from Instance Group and it's needed to
+  perform `apply` twice. To avoid situations like this, please use this resource
+  with the lifecycle `update_triggered_by` method, with the passed Instance's ID.
+immutable: true
+create_verb: :POST
+create_url: projects/{{project}}/zones/{{zone}}/instanceGroups/{{instance_group}}/addInstances
+delete_verb: :POST
+delete_url: projects/{{project}}/zones/{{zone}}/instanceGroups/{{instance_group}}/removeInstances
+read_verb: :POST
+self_link: projects/{{project}}/zones/{{zone}}/instanceGroups/{{instance_group}}/listInstances
+identity:
+  - instance
+nested_query: !ruby/object:Api::Resource::NestedQuery
+  keys:
+    - items
+references: !ruby/object:Api::Resource::ReferenceLinks
+  guides:
+    'Add instances': 'https://cloud.google.com/compute/docs/reference/rest/v1/instanceGroups/addInstances'
+    'Remove instances': 'https://cloud.google.com/compute/docs/reference/rest/v1/instanceGroups/removeInstances'
+    'List instances': 'https://cloud.google.com/compute/docs/reference/rest/v1/instanceGroups/listInstances'
+  api: 'https://cloud.google.com/compute/docs/reference/rest/v1/instanceGroups'
+async: !ruby/object:Api::OpAsync
+  operation: !ruby/object:Api::OpAsync::Operation
+    kind: 'compute#operation'
+    path: 'name'
+    base_url: 'projects/{{project}}/zones/{{zone}}/operations/{{op_id}}'
+    wait_ms: 1000
+  result: !ruby/object:Api::OpAsync::Result
+    path: 'targetLink'
+  status: !ruby/object:Api::OpAsync::Status
+    path: 'status'
+    complete: 'DONE'
+    allowed:
+      - 'PENDING'
+      - 'RUNNING'
+      - 'DONE'
+  error: !ruby/object:Api::OpAsync::Error
+    path: 'error/errors'
+    message: 'message'
+id_format: '{{project}}/{{zone}}/{{instance_group}}/{{instance}}'
+mutex: instanceGroups/{{project}}/zones/{{zone}}/{{instance_group}}
+exclude_tgc: true
+examples:
+  - !ruby/object:Provider::Terraform::Examples
+    name: 'instance_group_membership'
+    skip_test: true
+    vars:
+      network_name: 'network'
+      instance_group_name: 'instance-group'
+      instance_name: 'instance'
+custom_code: !ruby/object:Provider::Terraform::CustomCode
+  pre_delete: templates/terraform/pre_delete/compute_instance_group_membership.go.erb
+  custom_import: templates/terraform/custom_import/compute_instance_group_membership.go.erb
+  decoder: templates/terraform/decoders/instance_group_membership.go.erb
+  encoder: templates/terraform/encoders/compute_instance_group_membership.go.erb
+parameters:
+  - !ruby/object:Api::Type::ResourceRef
+    name: 'zone'
+    resource: 'Zone'
+    imports: 'name'
+    description: 'A reference to the zone where the instance group resides.'
+    url_param_only: true
+    custom_expand: 'templates/terraform/custom_expand/resourceref_with_validation.go.erb'
+  - !ruby/object:Api::Type::ResourceRef
+    name: 'instanceGroup'
```

Comment:
One thing I noticed on the second pass: `instance` is passed as a self_link, while `instance_group` is passed as a name along with its zone. Did you consider passing both as a self_link?

This will be hard to change later, so I want to verify that we're happy with this specific interface. As a user, it might be a bit confusing to use a self_link for one parameter but not the other.

Reply:
I've wanted to keep it consistent with the other fine-grained resources like `network_endpoint_group`. The `instance_group` is a URL parameter and it's name is used in the URL. From the API perspective I've wanted to make it consistent